src/stores/room-list/algorithms/Algorithm.ts

```diff
@@ -230,7 +230,10 @@ export class Algorithm extends EventEmitter {
             // Cheaply clone the rooms so we can more easily do operations on the list.
             // We optimize our lookups by trying to reduce sample size as much as possible
             // to the rooms we know will be deduped by the Set.
-            const rooms = this.cachedRooms[tagId];
+            const rooms = this.cachedRooms[tagId].map(r => r); // cheap clone
```

Comment:
more readable as `[...this.cachedRooms[tagId]]` IMO

Reply:
the code style within the room list is to already clone this way, so would prefer to match it at least

Reply:
actually that's a partial lie. Breadcrumbs is different with its `.slice()`, but that's probably a bug more than anything.